### PR TITLE
Fix tests to run with winvfs/cfapi

### PR DESCRIPTION
**Pull request by erikjv**

### CODE REVIEW

File: test/testsyncengine.cpp [hunk 1]
```diff
@@ -381,7 +406,10 @@ private slots:
         QFETCH_GLOBAL(bool, filesAreDehydrated);
         QFETCH(bool, sameMtime);
         QFETCH(QByteArray, checksums);
-        QFETCH(int, expectedGET);
+        QFETCH(int, expectedGEThydrated);
+        QFETCH(int, expectedGETdehydrated);
+
+        int expectedGET = filesAreDehydrated ? expectedGETdehydrated : expectedGEThydrated;
```
Comment (TheOneRing) [inline test/testsyncengine.cpp:412]: I expected the difference in the gets in a get triggered by 
`fakeFolder.localModifier().setContents(QStringLiteral("A/a1"), a1size, 'C');`
But this houls result in a constant increase of 1.
Also I added a compare that disproofs my assumption.
```cpp
fakeFolder.localModifier().setContents(QStringLiteral("A/a1"), a1size, 'C');
QCOMPARE(counter.nGET, filesAreDehydrated ? 1 : 0);```

**Comment by TheOneRing** [comment]
(better rebase your  code)

**Comment by TheOneRing** [comment]
Ci is failing: https://github.com/owncloud/client-desktop-vfs-win/actions/runs/3411635792

**Comment by erikjv** [comment]
I'd like to merge this PR, and fix any other issues in other repositories after the merge.

**Pull request merged**